runtime/commonMain/src/kotlinx/serialization/Tagged.kt

```diff
@@ -122,12 +132,13 @@ abstract class TaggedEncoder<Tag : Any?> : Encoder, CompositeEncoder {
     final override fun encodeStringElement(desc: SerialDescriptor, index: Int, value: String) = encodeTaggedString(desc.getTag(index), value)
 
     final override fun <T : Any?> encodeSerializableElement(desc: SerialDescriptor, index: Int, serializer: SerializationStrategy<T>, value: T) {
-        if (encodeElement(desc, index))
+        if (encodeElement(desc, index, value))
             encodeSerializableValue(serializer, value)
     }
 
     final override fun <T : Any> encodeNullableSerializableElement(desc: SerialDescriptor, index: Int, serializer: SerializationStrategy<T>, value: T?) {
-        if (encodeElement(desc, index))
+        if (value == null && encodeNullElement(desc, index)
+            || value != null && encodeElement(desc, index, value))
```

Comment:
This is invalid. It violates the design approach of this class and doesn't separate out the different aspects. As such it makes the class design inconsistent.